ORBIT/phases/install/quayside_assembly_tow/common.py

```diff
@@ -0,0 +1,305 @@
+        section at quayside.
+        """
+
+        yield self.task(
+            "Lift and Attach Tower Section",
+            12,
+            constraints={"windspeed": le(15)},
+        )
+
+    @process
+    def lift_and_attach_nacelle(self):
+        """
+        Task representing time associated with lifting and attaching a nacelle
+        at quayside.
+        """
+
+        yield self.task(
+            "Lift and Attach Nacelle", 7, constraints={"windspeed": le(15)}
+        )
+
+    @process
+    def lift_and_attach_blade(self):
+        """
+        Task representing time associated with lifting and attaching a turbine
+        blade at quayside.
+        """
+
+        yield self.task(
+            "Lift and Attach Blade", 3.5, constraints={"windspeed": le(12)}
+        )
+
+    @process
+    def mechanical_completion(self):
+        """
+        Task representing time associated with performing mechanical compltion
+        work at quayside.
+        """
+
+        yield self.task(
+            "Mechanical Completion", 24, constraints={"windspeed": le(18)}
+        )
+
+
+class TowingGroup(Agent):
+    """Class to represent an arbitrary group of towing vessels."""
+
+    def __init__(self, vessel_specs, num=1):
+        """
+        Creates an instance of TowingGroup.
+
+        Parameters
+        ----------
+        vessel_specs : dict
+            Specs for the individual vessels used in the towing group.
+            Currently restricted to one vessel specification per group.
+        """
+
+        super().__init__(f"Towing Group {num}")
+        self._specs = vessel_specs
+        self.day_rate = self._specs["vessel_specs"]["day_rate"]
+
+    def initialize(self):
+        """Initializes the towing group."""
+
+        self.submit_debug_log(message="{self.name} initialized.")
+
+    @process
+    def group_task(self, name, duration, num_vessels, **kwargs):
+        """
+        Submits a group task with any number of towing vessels.
+
+        Parameters
+        ----------
+        name : str
+            Name of task to complete. Used for submitting action logs.
+        duration : float | int
+            Duration of the task.
+            Rounded up to the nearest int.
+        num_vessels : int
+            Number of individual towing vessels needed for the operation.
+        """
+
+        kwargs = {**kwargs, "num_vessels": num_vessels}
+        yield self.task(name, duration, **kwargs)
+
+    def operation_cost(self, hours, **kwargs):
+        """
+        Returns cost of an operation of duration `hours` using number of
+        vessels, `num_vessels`.
+
+        Parameters
+        ----------
+        hours : int | float
+            Duration of operation in hours.
+        vessels : int
+            Default: 1
+        """
+
+        mult = kwargs.get("cost_multiplier", 1.0)
+        vessels = kwargs.get("num_vessels", 1)
+        return (self.day_rate / 24) * vessels * hours * mult
```

Comment:
Here, is self.day_rate the aggregated day rate of all the vessels in teh group?

Reply:
This attribute is the vessel day rate for one vessel but there are specialized methods that will return a "total" day rate for the group dependent on the number of vessels needed for each operation.